spec/requests/potepan/products_spec.rb

```diff
@@ -4,19 +4,43 @@
 RSpec.describe 'Potepan::Products', type: :request do
   describe 'GET #show' do
     let(:product) { create(:product) }
+    let(:taxon) { create(:taxon) }
+    let(:related_products_list) { create_list(:product, 5, taxons: [taxon]) }
+    let(:related_products_addition) { create(:product, taxons:[taxon], price: "20.20") }
+    let(:related_products) { [related_products_list,related_products_addition].flatten }
 
     before do
+      related_products.each { |related_product| related_product.images << create(:image) }
+      product.images.each { |image| image << create(:image) }
```

Comment:
imageは画像のはずですが、`<<`配列のように取り扱っているのはなぜでしょう。

Reply:
<<を＝に変更すれば良いという認識で大丈夫でしょうか？

Reply:
> imageは画像のはずですが、`<<`配列のように取り扱っているのはなぜでしょう。

<<を＝に変更すれば良いという認識で大丈夫でしょうか？

Reply:
違います。
ブロック引数の`image`は「単一の」画像オブジェクトを指しているため、それに対して`<<`演算子で新しい画像を追加したところで元の商品オブジェクト(`product`)に紐づく画像には何ら変化をもたらさない、ということです。

`product`に紐付いている画像対して破壊的な変更(新しい画像を追加)を行うには下記のようにする必要があります。
```suggestion
      product.images << create(:image)
```

Reply:
結論として、feature specの方で対応しました。
URL
理由として、requests/specの方で対応しようとしましたが　imageのチェックがソース属性でしか判断する内容がないと思い、画面のテストにて画像が登録したカウント通り表示されていることを確認することで担保しました。